pkg/mcs/tso/server/server.go

```diff
@@ -256,15 +444,180 @@ func checkStream(streamCtx context.Context, cancel context.CancelFunc, done chan
+		log.Info("stopping grpc gracefully is taking longer than expected and force stopping now")
+		gs.Stop()
+	}
+
+	if s.IsClosed() {
+		log.Info("grpc server stopped")
+	} else {
+		log.Fatal("grpc server stopped unexpectedly", errs.ZapError(serverr))
+	}
+}
+
+func (s *Server) startHTTPServer(l net.Listener) {
+	defer s.serverLoopWg.Done()
+
+	handler, _ := SetUpRestHandler(s.service)
+	hs := &http.Server{
+		Handler:           handler,
+		ReadTimeout:       5 * time.Minute,
+		ReadHeaderTimeout: 5 * time.Second,
+	}
+	serverr := hs.Serve(l)
+	log.Info("http server stopped serving")
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultHTTPGracefulShutdownTimeout)
+	defer cancel()
+	if err := hs.Shutdown(ctx); err != nil {
+		log.Error("http server shutdown encountered problem", errs.ZapError(err))
+	} else {
+		log.Info("all http(s) requests finished")
+	}
+	if s.IsClosed() {
+		log.Info("http server stopped")
+	} else {
+		log.Fatal("http server stopped unexpectedly", errs.ZapError(serverr))
+	}
+}
+
+func (s *Server) startGRPCAndHTTPServers(l net.Listener) {
+	defer s.serverLoopWg.Done()
+
+	mux := cmux.New(l)
+	grpcL := mux.MatchWithWriters(cmux.HTTP2MatchHeaderFieldSendSettings("content-type", "application/grpc"))
+	httpL := mux.Match(cmux.Any())
+
+	s.serverLoopWg.Add(2)
+	go s.startGRPCServer(grpcL)
+	go s.startHTTPServer(httpL)
+
+	if err := mux.Serve(); err != nil {
+		if s.IsClosed() {
+			log.Info("mux stop serving", errs.ZapError(err))
+		} else {
+			log.Panic("mux stop serving unexpectedly", errs.ZapError(err))
+		}
+	}
+}
+
+func (s *Server) startServer() (err error) {
+	// TODO: uncomment the following code to generate a unique cluster id from the given tsoClusterIDPath
+	// after we add rpc for the client to retrieve the clusgter id from the server then use it in every
+	// request for verification.
+	// if s.clusterID, err = etcdutil.InitClusterID(s.etcdClient, tsoClusterIDPath); err != nil {
+	//	return err
+	// }
+	log.Info("init cluster id", zap.Uint64("cluster-id", s.clusterID))
+
+	// It may lose accuracy if use float64 to store uint64. So we store the cluster id in label.
+	metadataGauge.WithLabelValues(fmt.Sprintf("cluster%d", s.clusterID)).Set(0)
+	// The independent TSO service still reuses PD version info since PD and TSO are just
+	// different service modes provided by the same pd-server binary
+	serverInfo.WithLabelValues(versioninfo.PDReleaseVersion, versioninfo.PDGitHash).Set(float64(time.Now().Unix()))
+	s.rootPath = path.Join(tsoRootPath, strconv.FormatUint(s.clusterID, 10))
+
+	// TODO: Figure out how we should generated the unique id and name passed to Participant.
+	// For now, set the name to be listen address and generate the unique id from the name with sha256.
+	uniqueName := s.cfg.ListenAddr
+	hash := sha256.Sum256([]byte(uniqueName))
+	uniqueID := binary.LittleEndian.Uint64(hash[:8])
+	log.Info(fmt.Sprintf("Participant name=%s, id=%d", uniqueName, uniqueID))
```

Comment:
Please use the unified log style.

Reply:
Do you mean start with lowercase character? Changed to ""participant name=%s, id=%d""